lib/websocket.js

```diff
@@ -17,29 +17,120 @@ const OPCODE_SHORT = 0x81;
 const LEN_16_BIT = 126;
 const MAX_16_BIT = 65536;
 const LEN_64_BIT = 127;
+const MAX_64_BIT = 0x7fffffffffffffffn + 1n;
 
-const calcOffset = (frame, length) => {
-  if (length < LEN_16_BIT) return [2, 6];
-  if (length === LEN_16_BIT) return [4, 8];
-  return [10, 14];
+const DEFAULT_OPTIONS = {
+  server: true,
 };
 
-const parseFrame = (frame) => {
-  const length = frame[1] ^ 0x80;
-  const [maskOffset, dataOffset] = calcOffset(frame, length);
-  const mask = frame.subarray(maskOffset, maskOffset + MASK_LENGTH);
-  const data = frame.subarray(dataOffset);
-  return { mask, data };
-};
+const xor = (cond1, cond2) => (cond1 || cond2) && !(cond1 && cond2);
+
+class Frame {
+  #frame = null;
+  #mask = null;
+  #dataOffset = 6;
+  #length = 0n;
+  #server = false;
```

Comment:
Why server variable is required? The task was about refactoring, not about adding new features.